src/enemy/bowMoblin.c

```diff
@@ -1,37 +1,384 @@
+#define NENT_DEPRECATED
 #include "global.h"
 #include "entity.h"
 #include "enemy.h"
 #include "functions.h"
 
-extern void sub_0803C5F0(Entity*);
+typedef struct {
+    Entity base;
+    u8 filler[0x12];
+    u8 unk_0x7a;
+    u8 unk_0x7b;
+    u8 filter[0x4];
+    u8 unk_0x80;
+    u8 unk_0x81;
+    u8 unk_0x82;
+    u8 unk_0x83;
+} BowMoblinEntity;
+
+void sub_0803C5F0(BowMoblinEntity*);
+void sub_0803C690(BowMoblinEntity*);
+void sub_0803C4B0(BowMoblinEntity*);
+void sub_0803C5C4(BowMoblinEntity*);
+void sub_0803C624(BowMoblinEntity*);
+void sub_0803C6DC(BowMoblinEntity*);
+void sub_0803C714(BowMoblinEntity*);
+void sub_0803C634(BowMoblinEntity*);
+u32 sub_0803C6F8(BowMoblinEntity*);
+u32 sub_0803C568(BowMoblinEntity*);
+extern Entity* sub_08049DF4(u32);
+extern u32 sub_0806FC80(Entity*, Entity*, s32);
+void sub_0803C664(BowMoblinEntity*);
 
 extern void (*const gUnk_080CFF78[])(Entity*);
 extern void (*const gUnk_080CFF90[])(Entity*);
```

Comment:
Since you're using the new entity style, these function tables will want to take the specific entity type.
```suggestion
extern void (*const gUnk_080CFF78[])(BowMoblinEntity*);
extern void (*const gUnk_080CFF90[])(BowMoblinEntity*);
```